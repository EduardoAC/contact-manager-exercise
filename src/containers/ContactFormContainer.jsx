import React from 'react';
import PropTypes from 'prop-types';

const ContactFormContainer = (props) => {
  const { handleSubmit } = props;
  return (
    <form onSubmit={handleSubmit} >
      <p>First name:</p>
      <input type="text" name="firstName" />
      <p>Last name:</p>
      <input type="text" name="lastName" />
      <p>Address:</p>
      <input type="text" name="address" />
      <p>Phone number: </p>
      <input type="text" name="phoneNumber" />
      <p>email: </p>
      <input type="text" name="email" />
      <input type="submit" value="Submit" />
    </form>
  );
}

ContactFormContainer.propTypes = {
  handleSubmit: PropTypes.func.isRequired
}

export default ContactFormContainer;
