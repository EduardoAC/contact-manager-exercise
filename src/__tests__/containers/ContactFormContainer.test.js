import React from 'react';
import { shallow, mount } from 'enzyme';

import ContactFormContainer from '../../containers/ContactFormContainer';

describe('<ContactFormContainer />', () => {
  it('Component redenrs correctly', () => {
    const wrapper = shallow(<ContactFormContainer handleSubmit={jest.fn()}/>);
    expect(wrapper).toMatchSnapshot();
  });
  it('should submit the form', () => {
    const handleSubmit = jest.fn();
    const wrapper = mount(<ContactFormContainer handleSubmit={handleSubmit}/>);
    const submitButton = wrapper.find("input[type='submit']");
    submitButton.simulate('submit');
    expect(handleSubmit).toHaveBeenCalledTimes(1);
    // Waiting for redux-form
    // expect(handleSubmit).toHaveBeenCalledWith({
    //   firstName: 'Homer',
    //   lastName: 'Simpson',
    //   address: 'Sprinfield',
    //   phoneNumber: '555-sprinfield',
    //   email: 'homer.simpson@sprinfield.com'
    // });
  });
});
