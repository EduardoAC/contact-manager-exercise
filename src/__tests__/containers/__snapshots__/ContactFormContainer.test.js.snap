// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactFormContainer /> Component redenrs correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <form
    onSubmit={[Function]}
>
    <p>
        First name:
    </p>
    <input
        name="firstName"
        type="text"
    />
    <p>
        Last name:
    </p>
    <input
        name="lastName"
        type="text"
    />
    <p>
        Address:
    </p>
    <input
        name="address"
        type="text"
    />
    <p>
        Phone number: 
    </p>
    <input
        name="phoneNumber"
        type="text"
    />
    <p>
        email: 
    </p>
    <input
        name="email"
        type="text"
    />
    <input
        type="submit"
        value="Submit"
    />
</form>,
  "nodes": Array [
    <form
      onSubmit={[Function]}
>
      <p>
            First name:
      </p>
      <input
            name="firstName"
            type="text"
      />
      <p>
            Last name:
      </p>
      <input
            name="lastName"
            type="text"
      />
      <p>
            Address:
      </p>
      <input
            name="address"
            type="text"
      />
      <p>
            Phone number: 
      </p>
      <input
            name="phoneNumber"
            type="text"
      />
      <p>
            email: 
      </p>
      <input
            name="email"
            type="text"
      />
      <input
            type="submit"
            value="Submit"
      />
</form>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <ContactFormContainer
        handleSubmit={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "handleSubmit": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <form
          onSubmit={[Function]}
>
          <p>
                    First name:
          </p>
          <input
                    name="firstName"
                    type="text"
          />
          <p>
                    Last name:
          </p>
          <input
                    name="lastName"
                    type="text"
          />
          <p>
                    Address:
          </p>
          <input
                    name="address"
                    type="text"
          />
          <p>
                    Phone number: 
          </p>
          <input
                    name="phoneNumber"
                    type="text"
          />
          <p>
                    email: 
          </p>
          <input
                    name="email"
                    type="text"
          />
          <input
                    type="submit"
                    value="Submit"
          />
</form>,
        "_debugID": 2,
        "_renderedOutput": <form
          onSubmit={[Function]}
>
          <p>
                    First name:
          </p>
          <input
                    name="firstName"
                    type="text"
          />
          <p>
                    Last name:
          </p>
          <input
                    name="lastName"
                    type="text"
          />
          <p>
                    Address:
          </p>
          <input
                    name="address"
                    type="text"
          />
          <p>
                    Phone number: 
          </p>
          <input
                    name="phoneNumber"
                    type="text"
          />
          <p>
                    email: 
          </p>
          <input
                    name="email"
                    type="text"
          />
          <input
                    type="submit"
                    value="Submit"
          />
</form>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ContactFormContainer
    handleSubmit={[Function]}
/>,
}
`;
